src/js/base/ScrollList.tsx

```diff
@@ -39,16 +41,23 @@ export function ScrollList({
     isLoading,
     items,
     renderRow,
+    hasNextPage,
 }: ScrollListProps) {
     useEffect(() => {
-        function onScroll() {
+        const isScrollEnabled = document.documentElement.scrollHeight > window.innerHeight;
+
+        if (!isScrollEnabled && hasNextPage) {
+            handleFetchNextPage();
+        }
```

Comment:
```suggestion
        handleFetchNextPage()
```
? (probably also a good idea to move calling the function below the functions defn, java-script will hoist in many cases, but I don't like relying on it)

Reply:
Removing the if statement and just having the `handleFetchNextPage() function called loads all pages on initial render even if scroll is enabled

Reply:
Ah interesting, do you know why? I would have thought that as more pages are loaded the scroll percentage would drop preventing further loading.

Reply:
Seems like the `getScrollRatio` consistently returns 1 so it basically spams the `fetchNextPage`. Not sure if this is due to the fact that i'm zoomed out or not though